components/file_upload/index.js

```diff
@@ -23,4 +23,4 @@ function mapStateToProps(state) {
     };
 }
 
-export default connect(mapStateToProps)(FileUpload);
+export default connect(mapStateToProps, null, null, {withRef: true})(FileUpload);
```

Comment:
Ah, so that's how to solve this problem. Last time I ran into an issue like this, I opted to lift the state, but I can see how there are cases where this is more complicated. Neat 👍

Reply:
This is just a quick hack after `connect` and `injectIntl` wrapped the component. I wish there's something better to approach this, especially because I couldn't last looking at:
```
this.refs.fileUpload.getWrappedInstance().refs.FileUpload.getWrappedInstance().cancelUpload(id);
```